api/control/v1alpha1/conditions.go

```diff
@@ -76,6 +78,18 @@ func SetCondition(conditions []BindingPolicyCondition, newCondition BindingPolic
 	return conditions
 }
 
+// UnsetCondition unsets the BindingPolicyCondition of the supplied type in
+// the given slice, if the BindingPolicyCondition of that type exists.
+func UnsetCondition(conditions []BindingPolicyCondition, conditionType ConditionType) []BindingPolicyCondition {
```

Comment:
I do not see a need for this func. Whenever the controller has an opinion about a given type of condition for a given object, it should ensure the presence of a Condition with that type and the proper values.